### PR TITLE
Drop deprecated e2e and update Readme

### PR DESCRIPTION
**Pull request by enxebre**
See:
https://github.com/openshift/machine-api-operator/pull/234
https://github.com/openshift/cluster-api-provider-aws/pull/170
https://github.com/openshift/kubernetes-autoscaler/pull/43

### CODE REVIEW

File: pkg/e2e/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ## Overview
 
-This e2e tests provide a mechanism to test end-to-end the behavior of the components of the system (i.e Openshift) which relate in any manner to the machine API domain.
+This e2e tests provide a mechanism to validate OpenShift/machine-API conformance user stories regardless the implementation details.
```
Comment (frobware) [inline pkg/e2e/README.md]: nit: regardless "of" the ...

File: pkg/e2e/README.md [hunk 2]
```diff
@@ -18,6 +18,8 @@ https://github.com/openshift/autoscaler
 
 ## Goals
 
+- Validate Openshift/machine-API conformance user stories regardless the implementation details
```
Comment (frobware) [inline pkg/e2e/README.md]: nit: regardless "of" the ...

File: pkg/e2e/README.md [hunk 3]
```diff
@@ -32,9 +34,9 @@ https://github.com/openshift/autoscaler
 
 ## Implementation
 
-Currently pure golang is used for running suite expectations but this can be moved to a framework eg. ginkgo as requirement rises
+We use [ginkgo](https://onsi.github.io/ginkgo/)
 
-This tests assume the existence of the system to be validated, i.e an Openshift cluster and a Kubeconfig file
+This tests assume the existence of the system to be validated, i.e an OpenShift cluster and a Kubeconfig file
```
Comment (frobware) [inline pkg/e2e/README.md]: nit: "These" or "The tests"

**Comment by enxebre** [comment]
thanks! @frobware addressed your comments

**Pull request merged**